### PR TITLE
feat: add error builder to options

### PR DESCRIPTION
**Pull request by FlutterJoey**

### CODE REVIEW

File: packages/flutter_availability/lib/src/config/availability_options.dart [hunk 1]
```diff
@@ -82,6 +85,16 @@ class AvailabilityOptions {
   /// which shows a platform adaptive loading indicator
   final WidgetBuilder loadingIndicatorBuilder;
 
+  /// Builder for handling errors.
+  ///
+  /// The user should determine how to display the error, through a dialog,
+  /// popup or other method of user interaction.
+  ///
+  /// There is not guarantee that there is a scaffold in the tree for each of
```
Comment (PureTryOut) [inline packages/flutter_availability/lib/src/config/availability_options.dart]: "no guarantee"

**Review by PureTryOut — CHANGES_REQUESTED** [review]
I'm probably missing something, but since `AvailabilityError` is not actually an error (but an enum), where is it being thrown? When is the new builder actually being called as I don't see that in the diff.

**Comment by FlutterJoey** [comment]
The error is an enum to have a single object type that has a clear list of possible options for the implementing size to handle. The specific exceptions are for the UI to handle, and in some cases will defer that to the implementer by using one of the enums

**Comment by bart-iconica** [comment]
I realize the UI has to handle the exceptions, but when are they being _thrown_? How are they even thrown at all when it's not actually an exception?

**Comment by FlutterJoey** [comment]
They are not thrown. This is the groundwork for error handling, so me and @freekvandeven can work on it independently.

They should be used like this:

1. Service throws an exception.
2. UI catches exception.
3. UI calls the builder with the specific error we want the client to build a visualisation for.
4. Implementation gets called

**Review by PureTryOut — APPROVED** [review]

**Pull request merged**